sql/core/src/main/scala/org/apache/spark/sql/execution/basicPhysicalOperators.scala

```diff
@@ -225,6 +226,57 @@ case class FilterExec(condition: Expression, child: SparkPlan)
   override def outputPartitioning: Partitioning = child.outputPartitioning
 }
 
+/** Physical plan for RecursiveTable. */
+case class RecursiveTableExec(
+    name: String,
+    anchorTerm: SparkPlan,
+    recursiveTerm: SparkPlan) extends SparkPlan {
+  override def children: Seq[SparkPlan] = Seq(anchorTerm, recursiveTerm)
+
+  override def output: Seq[Attribute] = anchorTerm.output
+
+  override val doNotPrepareInAdvance: Seq[SparkPlan] = Seq(recursiveTerm)
+
+  override protected def doExecute(): RDD[InternalRow] = {
+    var temp = anchorTerm.execute().map(_.copy()).cache()
+    var tempCount = temp.count()
+    var result = temp
+    var level = 0
+    do {
+      if (level > conf.getConf(SQLConf.RECURSION_LEVEL_LIMIT)) {
```

Comment:
shall we get it into a val member of the node in order to avoid getting it multiple times during execution?